relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -56,28 +56,44 @@ pub struct TransactionMetricsConfig {
 const METRIC_NAMESPACE: &str = "transactions";
 
 #[cfg(feature = "processing")]
-fn extract_transaction_status(transaction: &Event) -> Option<String> {
-    use relay_general::{
-        protocol::{Context, ContextInner},
-        types::Annotated,
-    };
+fn get_trace_context(event: &Event) -> Option<&TraceContext> {
+    let contexts = event.contexts.value()?;
+    let trace = contexts.get("trace").map(Annotated::value);
+    if let Some(Some(ContextInner(Context::Trace(trace_context)))) = trace {
+        return Some(trace_context.as_ref());
+    }
```

Comment:
I'm going to add a very similar method here: https://github.com/getsentry/relay/pull/1231 -- and some other similar changes

you can merge first, I will rebase my pr then